src/OpenLoco/src/GameCommands/CreateIndustry.cpp

```diff
@@ -0,0 +1,722 @@
+        const auto randVal = newIndustry->prng.randNext() & 0xFF;
+
+        auto prodRateRand = randVal;
+        for (auto i = 0; i < 2; ++i)
+        {
+            newIndustry->var_17D[i] = 0;
+            newIndustry->productionRate[i] = (((indObj->initialProductionRate[i].max - indObj->initialProductionRate[i].min) * prodRateRand) / 256) + indObj->initialProductionRate[i].min;
+            if (isEditorMode())
+            {
+                newIndustry->var_17D[i] = newIndustry->productionRate[i];
+                newIndustry->producedCargoQuantityPreviousMonth[i] = newIndustry->var_17D[i] * 30;
+            }
+            // This is odd but follows vanilla
+            prodRateRand = newIndustry->productionRate[i] & 0xFF;
+        }
+
+        currency32_t totalCost = 0;
+
+        // 0x00454552
+        const auto numBuildings = (((indObj->maxNumBuildings - indObj->minNumBuildings + 1) * randVal) / 256) + indObj->minNumBuildings;
+        for (auto i = 0U; i < numBuildings; ++i)
+        {
+            const auto building = indObj->buildings[i];
+            // 0x00E0C3D2 (bit 0)
+            const bool isMultiTile = indObj->buildingSizeFlags & (1ULL << building);
+
+            bool hasBuildingPlaced = false;
+            // Attempt to place
+            for (auto j = 0U; j < 25; ++j)
+            {
+                const auto randVal2 = newIndustry->prng.randNext();
+                // bh
+                const uint8_t direction = indObj->hasFlags(IndustryObjectFlags::notRotatable) ? 0 : randVal2 & 0x3;
+
+                const auto [distanceRange, minDistance] = kPlacementAttemptDistanceRangeAndMin[j];
+
+                const auto randTileX = static_cast<tile_coord_t>((distanceRange * ((randVal2 >> 18) & 0xFF)) / 256) + minDistance;
+                const auto randTileY = static_cast<tile_coord_t>((distanceRange * ((randVal2 >> 2) & 0xFF)) / 256) + minDistance;
+
+                const auto randPos = lastPlacedBuildingPos + World::toWorldSpace(World::TilePos2(randTileX, randTileY));
+                const auto minRandPos = randPos;
+                const auto maxRandPos = isMultiTile ? randPos + World::toWorldSpace(World::TilePos2{ 1, 1 }) : randPos;
+
+                // TODO: Yank this out into a function
+                bool placementFailure = false;
+                for (auto k = 0U; k < newIndustry->numTiles; ++k)
+                {
+                    const auto minTileLoc = World::Pos2{ newIndustry->tiles[k].x, newIndustry->tiles[k].y };
+                    const auto maxTileLoc = (newIndustry->tiles[k].z & (1U << 15)) ? minTileLoc + World::toWorldSpace(World::TilePos2{ 1, 1 }) : minTileLoc;
+
+                    if (minRandPos.x > maxTileLoc.x)
+                    {
+                        continue;
+                    }
+                    if (minRandPos.y > maxTileLoc.y)
+                    {
+                        continue;
+                    }
```

Comment:
Can we use the > operator for the Pos2 vectors instead?

Reply:
I'm not a fan of > on a Pos2.